src/main.rs

```diff
@@ -50,10 +50,64 @@ fn main() {
     }
 }
 
+fn build_cache_without_training(cl: clap::ArgMatches) -> Result<(), Box<dyn Error>> {
+    // We'll parse once the command line into cl and then different objects will examine it
+    let input_filename = cl.value_of("data").expect("--data expected");
+    let vw_namespace_map_filepath = Path::new(input_filename)
+        .parent()
+        .expect("Couldn't access path given by --data")
+        .join("vw_namespace_map.csv");
+    let vw: vwmap::VwNamespaceMap;
+    vw = vwmap::VwNamespaceMap::new_from_csv_filepath(vw_namespace_map_filepath)?;
+    let mut cache = cache::RecordCache::new(input_filename, cl.is_present("cache"), &vw);
+    let input = File::open(input_filename)?;
+    let mut aa;
```

Comment:
I'd replace variable names `aa` and `bb` with something more idiomatic, perhaps `raw_buffer` and `uncompressed_buffer`

Reply:
done